docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+-   The **principal encoded in the subaccount is the intended controller** of the created canister.
+-   Only that principal is authorized to call `notify_create_canister` and can set **arbitrary canister settings**, including additional controllers.
+-   **Exception**: The [NNS dapp](https://nns.ic0.app/) is permitted to call `notify_create_canister` on behalf of users and acts as a proxy in that case.
+-   The **memo field** must explicitly indicate the intent to create a canister. This can be expressed as:
+    -   A legacy 64-bit unsigned integer memo with value (in decimal):
+        ```
+        1095062083
+        ```
+    -   Or, for ICRC-1-compatible transfers (e.g., `icrc1_transfer`, `icrc2_transfer_from`), a memo blob equal to:
+        ```
+        "\43\52\45\41\00\00\00\00"  // ASCII "CREA" + 4 zero bytes
+        ```
+
+#### Parameters
+
+```candid
+record {
+  block_index: nat64;              // The ledger block containing the ICP payment
+  controller: principal;           // The creator of the canister. Must match caller; otherwise, Err is returned (unless caller is NNS Dapp). This is also used when checking that the creator is authorized to create canisters in subnets where authorization is required. This is also used when `settings` does not specify `controllers`.
+  settings: opt CanisterSettings;  // Optional settings like controllers, memory limits, etc.
+  subnet_type: opt text;           // Deprecated: legacy subnet selection
+  subnet_selection: opt SubnetSelection; // Preferred subnet selection method
+}
+```
+
+#### Returns
+
+```candid
+variant { Ok: principal; Err: NotifyError }
+```
+
+-   **Ok**: The principal of the newly created canister.
+-   **Err**: Indicates why the request was rejected (e.g., incorrect memo, controller mismatch, payment already processed, etc.).
+    For information about retry semantics, error handling, and caching behavior, see [Notify Methods – Shared Behavior](#notify-methods--shared-behavior-retry-error-handling-and-caching).
+
+## `notify_top_up`
+
+Tops up an existing canister by minting cycles based on an ICP payment recorded in the ICP Ledger.
+
+The CMC expects the payment to follow a strict structure that encodes both the **intent** and the **target canister**:
+
+-   The **destination account** of the ICP transfer must be the CMC's account with a subaccount derived from the target canister’s principal (see the section on [Encoding a principal into a subaccount](#encoding-a-principal-into-a-subaccount)).
+-   The **memo field** must explicitly indicate the intent to top up a canister. This can be expressed as:
+    -   A legacy 64-bit unsigned integer memo with value (in decimal)
+        ```
+        1347768404
+        ```
+    -   Or, for ICRC-1-compatible transfers (e.g., `icrc1_transfer`, `icrc2_transfer_from`), a memo blob equal to:
+        ```
+        "\50\55\50\54\00\00\00\00" // ASCII "TUPT" + 4 zero bytes
```

Comment:
```suggestion
        "\50\55\50\54\00\00\00\00" // ASCII "TPUP" + 4 zero bytes
```

Reply:
Hex may also need to be updated but it actually looks right...

`pub const MEMO_TOP_UP_CANISTER: Memo = Memo(0x50555054); // == 'TPUP'`